### ISSUE TITLE
Interfaces describing json schema shape

### ISSUE THREAD

**Issue by artonio** [guillotinaweb/ngx-schema-form#333]
Hi, first of all I would like to thank the community for this library, excellent work. However, one thing that I usually run into when developing software with this library is that it's in TypeScript but there are no interfaces describing the shape of the JSON Schema, i never know what keys it can have without resorting to documentation. It would be nice to have interfaces describing the schema as part of the library. Maybe something like this?

```
export interface IWidget {
	id?: string;
	plugins?: string;
	toolbar?: string;
}

export interface IFieldSet {
	id: string;
	title: string;
	fields: string[];
}

export interface IButton {
	id: string;
	label: string;
	widget?: string;
}

export interface IProperties {
	[prop: string]: IJsonSchema;
}

export interface IJsonSchema {
	type: 'object' | 'array' | 'string' | 'number' | 'integer' | 'boolean';
	id?: string;
	title?: string;
	description?: string;
	placeholder?: string;
	minLength?: number;
	maxLength?: number;
	visibleIf?: any;
	enum?: any[];
	properties?: IProperties;
	buttons?: IButton[];
	required?: string[];
	isRoot?: boolean;
	widget?: IWidget | string;
	fieldsets?: IFieldSet[];
}
```

Let me know if it's a good idea and i could make a pull request potentially?

Thanks!

**Comment by ebrehault** [comment]
It would be great!!

**Comment by ebrehault** [comment]
Closed by #334
